packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -3274,6 +3299,51 @@ class SemanticsOwner extends ChangeNotifier {
 
   /// Update the semantics using [onSemanticsUpdate].
   void sendSemanticsUpdate() {
+    // Once the tree is up-to-date, verify that there's no invisible nodes in it.
+    assert(() {
+      final List<SemanticsNode> invisibleNodes = <SemanticsNode>[];
+      bool findInvisibleNodes(SemanticsNode rootNode) {
```

Comment:
Maybe rename the name rootNode to just node here? It gets a little confusing because this is not necessarily gonna be the root root node for every call?